source/extensions/filters/http/local_ratelimit/config.cc

```diff
@@ -18,8 +18,10 @@ Http::FilterFactoryCb LocalRateLimitFilterConfig::createFilterFactoryFromProtoTy
     const std::string&, Server::Configuration::FactoryContext& context) {
   FilterConfigSharedPtr filter_config = std::make_shared<FilterConfig>(
       proto_config, context.localInfo(), context.dispatcher(), context.scope(), context.runtime());
-  return [filter_config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
-    callbacks.addStreamFilter(std::make_shared<Filter>(filter_config));
+  Event::Dispatcher& dispatcher = context.dispatcher();
+
+  return [filter_config, &dispatcher](Http::FilterChainFactoryCallbacks& callbacks) -> void {
```

Comment:
`filter_config` takes the dispatcher as an argument to its constructer, why do we need to pass it here again? Let's just have filters access the dispatcher via `filter_config`.

Reply:
That's exactly what I had initially as well, i.e. accessing the dispatcher via the `filter_config`, only to realize that we are using a `const` pointer to the `filter_config` and as a result accessing non-const members/methods wouldn't work on it (dispatcher needs to be non-const).

Reply:
You can use `mutable`...

Reply:
oh you mean `getConfig()` here:

https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/local_ratelimit/local_ratelimit.h#L110

that's fine, it doesn't have to be const we can drop that....

Reply:
Actually, don't drop the `const` in `getConfig()` -- it's fine to return a non const reference from a const method:

https://godbolt.org/z/P6vvarhn3

Reply:
Turns out the event `dispatcher` from `FilterConfig` isn't thread safe for use from within the stream filter which executes in a different thread. So I switched to using the thread local `dispatcher` instead.